doc/man3/flux_kvs_lookup.rst

```diff
@@ -155,6 +155,17 @@ FLUX_KVS_WATCH_APPEND
    response will only contain the additional appended data.  If the
    value is overwritten, the lookup fails with EINVAL.
 
+FLUX_KVS_WATCH_INITIAL_SENTINEL
+   Specified along with FLUX_KVS_WATCH_APPEND, this flag will result
+   in an empty RPC message to be sent when all "initial" or "current"
+   KVS data has been sent.  This empty RPC message serves as a
+   sentinel and is useful for identifying what KVS data is "new" vs
+   "old" at the point in time a :var:`key` starts to be watched.
+   Normally an empty RPC message will result in EPROTO errors, but
+   :func:`flux_kvs_lookup_get` and :func:`flux_kvs_lookup_get_raw`
+   will recognize the sentinel RPC and return NULL / zero for its
+   results.
+
```

Comment:
It seems like this description is a bit focused on the internal protocol.  Here's a stab at rewriting it to be more API focused:
> Modifies the behavior of the FLUX_KVS_WATCH_APPEND flag to notify the caller once
> all the data that has been appended to the key thus far has been received.
> flux_kvs_lookup_get()  indicates this by returning success with a NULL value parameter.